Samples/Widgets/cpp-win32-packaged/SampleWidgetProviderApp/WidgetImplBase.h

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+class WidgetImplBase
```

Comment:
since you are using this type in shared pointers, please have the type inherit from `std::enable_shared_from_this<WidgetImplBase>`
Otherwise, bad things might happen when two shared pointers don't share the same reference count, see notes in https://en.cppreference.com/w/cpp/memory/enable_shared_from_this